apps/src/code-studio/components/header/EditableProjectName.jsx

```diff
@@ -0,0 +1,141 @@
+/* globals dashboard */
+
+import PropTypes from 'prop-types';
+import React from 'react';
+import i18n from '@cdo/locale';
+import {connect} from 'react-redux';
+
+import ProjectUpdatedAt from './ProjectUpdatedAt';
+import {refreshProjectName} from '../../headerRedux';
+
+const styles = {
+  buttonWrapper: {
+    float: 'left'
+  }
+};
+
+class UnconnectedDisplayProjectName extends React.Component {
+  static propTypes = {
+    beginEdit: PropTypes.func.isRequired,
+    projectName: PropTypes.string.isRequired
+  };
+
+  render() {
+    return (
+      <div style={styles.buttonWrapper}>
+        <div className="project_name_wrapper header_text">
+          <div className="project_name header_text">
+            {this.props.projectName}
+          </div>
+          <ProjectUpdatedAt />
+        </div>
+        <div
+          className="project_edit header_button header_button_light"
+          onClick={this.props.beginEdit}
+        >
+          {i18n.rename()}
+        </div>
+      </div>
+    );
+  }
+}
+const DisplayProjectName = connect(state => ({
+  projectName: state.header.projectName
+}))(UnconnectedDisplayProjectName);
+
+class UnconnectedEditProjectName extends React.Component {
+  static propTypes = {
+    finishEdit: PropTypes.func.isRequired,
+    projectName: PropTypes.string.isRequired,
+    refreshProjectName: PropTypes.func.isRequired
+  };
+
+  state = {
+    savingName: false
+  };
+
+  saveNameChange = () => {
+    if (this.state.savingName) {
+      return;
+    }
+
+    const newName = this.nameChangeInput.value.trim().substr(0, 100);
+    if (newName.length === 0) {
+      return;
+    }
+
+    dashboard.project.rename(newName, () => {
+      dashboard.header.updateTimestamp();
```

Comment:
Possible next steps: Can we remove these global dependencies by just importing [project.js](https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/code-studio/initApp/project.js) and [header.js](https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/code-studio/header.js) directly?

Reply:
oh, I had been thinking of reworking all of that to use redux internally, but just importing the files is _way_ simpler!

Tracking [here](https://codedotorg.atlassian.net/browse/FND-489)